tla-bmcmt/src/main/scala/at/forsyte/apalache/tla/bmcmt/rules/MkSeqRule.scala

```diff
@@ -68,19 +67,46 @@ class MkSeqRule(rewriter: SymbStateRewriter) extends RewritingRule {
       throw new RewriterException("%s is not applicable".format(getClass.getSimpleName), state.ex)
   }
 
-  // extract capacity from the length expression
-  private def extractCapacity(lenEx: TlaEx): Int = {
-    lenEx match {
-      case ValEx(TlaInt(n)) if n.isValidInt && n >= 0 =>
-        n.toInt
+  // Extract capacity from a constant expression that may contain a call to __ApalacheSeqCapacity.
+  // The main use for this method is internal rewiring, e.g., in `__rewire_sequences_ext_in_apalache.tla`.
+  // This code somewhat duplicates ConstSimplifier.
+  private def computeCapacity: (SymbState, TlaEx) => (SymbState, Int) = {
+    case (state, ValEx(TlaInt(n))) if n.isValidInt && n >= 0 =>
+      (state, n.toInt)
+
+    case (_, e @ ValEx(TlaInt(n))) if !n.isValidInt || n < 0 =>
+      val msg = s"Expected an integer in the range [0, ${Int.MaxValue}]. Found: $n"
+      throw new TlaInputError(msg, Some(e.ID))
+
+    case (state, OperEx(ApalacheInternalOper.apalacheSeqCapacity, seq)) =>
+      // rewrite the sequence expression and extract its constant capacity
+      val nextState = rewriter.rewriteUntilDone(state.setRex(seq))
+      val (_, _, capacity) = proto.unpackSeq(nextState.arena, nextState.asCell)
+      (nextState, capacity)
+
+    case (state, OperEx(TlaArithOper.plus, left, right)) =>
+      val (lstate, lcap) = computeCapacity(state, left)
+      val (rstate, rcap) = computeCapacity(lstate, right)
+      (rstate, lcap + rcap)
```

Comment:
`lcap + rcap` may overflow `lcap + rcap`?

Reply:
Technically, yes. But we will definitely never reach `MAX_INT`.